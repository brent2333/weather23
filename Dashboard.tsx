import { useContext, useState } from "react";
import { useQuery } from "@tanstack/react-query";

const Dashboard = () => {
  const [requestParams, setRequestParams] = useState("");


//   const results = useQuery(["search", requestParams], fetchSearch);


  return (
    <div className="">
      <form
        onSubmit={(event: MouseEvent): void => {
        event.preventDefault();
          const formData = new FormData(event.target as HTMLFormElement);
          const loc = formData.get("location")?.toString() ?? ""
          setRequestParams(loc);
        }}
      >

        <label htmlFor="location">
          Location
          <input id="location" name="location" placeholder="Location" />
        </label>


        <button>Submit</button>
      </form>
    </div>
  );
};

export default Dashboard;
