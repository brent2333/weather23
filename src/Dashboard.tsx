import { FormEvent, Fragment, lazy, Suspense } from "react";
import { useGetCurrentWeatherQuery } from "./weatherApiService";
import { useAppSelector, useAppDispatch } from "./hooks";
import { current } from "./weatherSearchSlice";
import { ArrowPathIcon } from "@heroicons/react/24/outline";

const Results = lazy(() => import("./Results"));
const Forecast = lazy(() => import("./Forecast"));


const Dashboard = () => {
  const selectedCurrentLocation = useAppSelector(
    (state) => state.weatherSearch.value.currentLocation
  );
  let { data: currentWeather } = useGetCurrentWeatherQuery(
    selectedCurrentLocation
  );
  currentWeather = currentWeather ?? undefined;
  const dispatch = useAppDispatch();

  return (
    <Fragment>
      <div className="col-auto lg:max-w-xs h-48 sm:min-w-full md:min-w-full">
        <form
          onSubmit={(event: FormEvent<HTMLFormElement>): void => {
            event.preventDefault();
            const formData = new FormData(event.target as HTMLFormElement);
            const loc = formData.get("location")?.toString() ?? "";
            dispatch(current(loc));
          }}
          className="bg-gray-300 shadow-md rounded px-8 pt-6 pb-6 m-6 lg:max-w-md md:max-w-sm sm:max-w-full"
        >
          <div className="mb-4">
            <label
              htmlFor="location"
              className="block text-gray-700 text-sm font-bold mb-2"
            >
              Search a Location
              <input
                id="location"
                name="location"
                placeholder="Location"
                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
              />
            </label>
          </div>
          <button className="inline-flex justify-center rounded-md border border-transparent bg-indigo-600 py-2 px-4 text-sm font-medium text-white shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2">
            Submit
          </button>
        </form>
      </div>
      {currentWeather ? (
        <div data-testid="results-container">
          <Suspense
              fallback={
                <div className="items-center justify-center flex p-4">
                  <div className="w-10 h-10 animate-spin"><ArrowPathIcon/></div>
                </div>
              }>
              <Results current={currentWeather} />
          </Suspense>
          <Suspense
              fallback={
                <div className="items-center justify-center flex p-4">
                  <div className="w-10 h-10 animate-spin"><ArrowPathIcon/></div>
                </div>
              }
            >
          <Forecast location={selectedCurrentLocation} />
          </Suspense>
        </div>
      ) : null}
    </Fragment>
  );
};

export default Dashboard;
