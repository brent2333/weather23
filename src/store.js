import { configureStore } from "@reduxjs/toolkit";
import { weatherApi } from './weatherApiService';
import weatherSearch from './weatherSearchSlice';
const store = configureStore({
    reducer: {
        weatherSearch,
        [weatherApi.reducerPath]: weatherApi.reducer
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(weatherApi.middleware)
})

export default store;