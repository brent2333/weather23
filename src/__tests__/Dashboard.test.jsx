import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import store from "../store";
import { Provider } from "react-redux";
import Dashboard from "../Dashboard";

test("dashboard renders on load", async () => {
    const { asFragment } = render(<Provider store={store}><Dashboard /></Provider>);
    expect(asFragment()).toMatchSnapshot();
})