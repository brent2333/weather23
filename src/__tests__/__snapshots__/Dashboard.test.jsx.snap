// Vitest Snapshot v1

exports[`dashboard renders on load 1`] = `
<DocumentFragment>
  <div
    class="col-auto lg:max-w-xs h-48 sm:min-w-full md:min-w-full"
  >
    <form
      class="bg-gray-300 shadow-md rounded px-8 pt-6 pb-6 m-6"
    >
      <div
        class="mb-4"
      >
        <label
          class="block text-gray-700 text-sm font-bold mb-2"
          for="location"
        >
          Search a Location
          <input
            class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            id="location"
            name="location"
            placeholder="Location"
          />
        </label>
      </div>
      <button
        class="inline-flex justify-center rounded-md border border-transparent bg-indigo-600 py-2 px-4 text-sm font-medium text-white shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
